tests/testthat/test-tcplhit2_core.R

```diff
@@ -13,3 +13,150 @@ test_that("tcplhit2 works", {
   expect_equal(output$tp, 0.749, tolerance = 1e-2)
   expect_equal(output$ga, 9.59, tolerance = 1e-2)
 })
+
+test_that("tcplhit2 BMD boundary check", {
+  # Preparation for tests on BMD boundary arguments
+  # Simulate some data
+  X <- rep(seq(0.001,3,length.out = 10),each = 5)
+
+  set.seed(918)
+  Case_1 <- hillfn(ps = c(4,0.00105,1.08,0.1),x = X) + rt(n = length(X),df = 4)
+  Case_3 <- pow(ps = c(10,5,0.1),x = X) + rt(n = length(X),df = 4)
+  set.seed(455)
+  Case_2 <- hillfn(ps = c(4,0.00105,1.08,0.1),x = X) + rt(n = length(X),df = 4)
+  set.seed(321)
+  Case_4 <- exp2(ps = c(5.6,15,0.1),x = X) + rt(n = length(X),df = 4)
+  set.seed(322)
+  Case_5 <- exp5(ps = c(4,10,5,0.1),x = X) + rt(n = length(X),df = 4)
+
+  Y <- rbind(Case_1, Case_2, Case_3, Case_4, Case_5)
+
+  # Y contains 5 rows of simulated responses, each row corresponds to
+  # one data cases in order:
+
+  # 1. BMD < lower threshold
+  # 2. lower threshold < BMD < lowest expt dose
+  # 3. lowest expt dose < BMD < upper expt dose
+  # 4. upper expt dose < BMD < upper threshold
+  # 5. BMD > upper threshold
+
+  df <- matrix(nrow = 5, ncol = 14)
+  colnames(df) <- c("onesd", "cutoff", "bmd", "bmdu", "bmdl",
+                    "combo1", "combo1u", "combo1l",
+                    "combo2", "combo2u", "combo2l",
+                    "combo3", "combo3u", "combo3l")
+
+  # fit each data case and hit-call with different argument combinations
+  for (i in 1:5) {
+    temp <- Y[i, 1:10]
+    bmad <- mad(temp)
+    onesd <- sd(temp)
+    cutoff <- 3*bmad
+
+    df[i, "onesd"] <- onesd
+    df[i, "cutoff"] <- cutoff
+
+    params <- tcplfit2_core(X, Y[i, ], cutoff,
+                            force.fit = T, bidirectional = F)
+    # No thresholds specified, no censoring
+    output <- tcplhit2_core(params, X, Y[i,], cutoff, onesd,
+                            bmed=0, conthits=T, aicc=F)
+
+    df[i, "bmd"] <- output$bmd
+    df[i, "bmdu"] <- output$bmdu
+    df[i, "bmdl"] <- output$bmdl
+
```

Comment:
Suggest putting a brief comment for the choice of bmd boundary multiples.  For example,
"# For these tests we use a lower and upper BMD threshold multiple of 0.7 and 10, respectively, to demonstrate use cases work for the simulated concentration-response data.  Threshold multiples are not necessarily meant to align with the typical suggested values of 0.1 and 10 for lower and upper threshold multiples, respectively."

Reply:
Comment added in the latest commit.

Reply:
See it now, thank you for confirming @gracezhihuizhao.